File: backend/src/main/java/com/dev/backend/service/CityService.java
package com.dev.backend.service;

import java.time.Instant;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.dev.backend.entities.City;
import com.dev.backend.repository.CityRepository;

@Service
public class CityService {
  
  @Autowired
  private CityRepository repository;  

  public List<City> findAll(){
    return repository.findAll();
  }

  public City insert(City obj){
    char firstLetter = obj.getName().charAt(0);
    String newName = String.valueOf(firstLetter).toUpperCase() + obj.getName().substring(1);
    obj.setName(newName);
    obj.setCreatDate(Instant.now());
    return repository.saveAndFlush(obj);
  }

  public City update(City obj){
    obj.setUpdateDate(Instant.now());
    return repository.saveAndFlush(obj);
  }

  public void delete(Long id) {
    repository.deleteById(id);
  }
}

File: frontend/sakai-js/demo/service/CityService.js
import axios from "axios";

export class CityService {
  url = 'http://localhost:8080/api/'

  findAll(){
    return axios.get(this.url+'cities')
  }

  insert(object){
    return axios.post(this.url+'cities', object)
  }

  update(object){
    return axios.put(this.url+'cities', object)
  }

  delete(id){
    return axios.delete(this.url+'cities/ '+ id)
  }
}